jdaviz/app.py

```diff
@@ -793,6 +821,13 @@ def _update_selected_data_items(self, viewer_id, selected_items):
                                                         sender=self)
                 self.hub.broadcast(remove_data_message)
 
+        # Sets the plot axes labels to be the units of the most recently
+        # active data.
+        if len(active_data_labels) > 0:
+            active_data = self.data_collection[active_data_labels[0]]
+            if type(active_data.get_object()) is Spectrum1D:
```

Comment:
So it seems that the failing tests are because `get_object` raises an error if no `_preferred_translation` exists for the object (I had assumed it'd just return the object as a Glue `Data` object if no translation was found). So to handle this, perhaps try something like:

```suggestion
            if active_data._preferred_object is not None and type(active_data.get_object()) is Spectrum1D:
```

Reply:
Got it, thanks! I changed those lines and I'll wait to see what the tests spit back out.